content/webapp/pages/search/index.tsx

```diff
@@ -305,59 +314,171 @@ const StoryPromoContainer = styled(Container)`
   const totalCatalogueResults = totalWorksResults + totalImagesResults;
   const totalResults = totalContentResults + totalCatalogueResults;
   const isSmallGallery =
-    catalogueResults.images && catalogueResults.images.results.length < 3;
+    clientSideImages && clientSideImages.results.length < 3;
   const pathname = usePathname();
 
+  useMemo(() => {
+    if (contentResultsLength !== contentResults?.totalResults)
+      setIsCatalogueLoading(true);
+    setContentResultsLength(contentResults?.totalResults);
+  }, [contentResults?.totalResults]);
+
+  async function fetchWorks() {
+    try {
+      setIsLoadingWorks(true);
+
+      const worksResults = await getWorks({
+        params: {
+          ...params,
+          aggregations: ['workType'],
+        },
+        pageSize: 1,
+        toggles: data.toggles,
+      });
+
+      const workTypeBuckets = getQueryWorkTypeBuckets({
+        categoryName: 'works',
+        queryResults: worksResults,
+      });
+
+      setClientSideWorkTypes(() => {
+        setIsLoadingWorks(false);
+        return workTypeBuckets;
+      });
+    } catch (e) {
+      setIsLoadingWorks(false);
+
+      return undefined;
+    }
+  }
+
+  async function fetchImages() {
+    try {
+      setIsLoadingImages(true);
+
+      const imagesResults = await getImages({
+        params,
+        pageSize: 7,
+        toggles: data.toggles,
+      });
+
+      const images = getQueryResults({
+        categoryName: 'images',
+        queryResults: imagesResults,
+      });
+
+      setClientSideImages(() => {
+        setIsLoadingImages(false);
+        return {
+          totalResults: images?.totalResults || 0,
+          results:
+            images?.pageResults.map(image => ({
+              ...image,
+              src: image.locations[0].url,
+              width: (image.aspectRatio || 1) * 100,
+              height: 100,
+            })) || [],
+          requestUrl: images?.requestUrl,
+        };
+      });
+    } catch (e) {
+      setIsLoadingImages(false);
+      return undefined;
+    }
+  }
+
   useEffect(() => {
-    if (apiToolbar) {
-      if (
-        catalogueResults.works &&
-        !extraApiToolbarLinks.find(link => link.id === 'catalogue-api-works')
-      ) {
-        setExtraApiToolbarLinks([
-          ...extraApiToolbarLinks,
-          {
-            id: 'catalogue-api-works',
-            label: 'Works API query',
-            link: catalogueResults.works.requestUrl,
-          },
-        ]);
-      }
+    const pathname = '/search';
+    setLink({
+      href: { pathname, query },
+      as: { pathname, query },
+    });
 
-      if (
-        catalogueResults.images?.requestUrl &&
-        !extraApiToolbarLinks.find(link => link.id === 'catalogue-api-images')
-      ) {
-        setExtraApiToolbarLinks([
-          ...extraApiToolbarLinks,
-          {
+    setIsCatalogueLoading(true);
+    setClientSideWorkTypes(undefined);
+    setClientSideImages(undefined);
```

Comment:
```suggestion
```
These were essentially doing what setIsCatalogueLoading  is now doing, so can get rid of them.

Reply:
[addressed in latest commit](https://github.com/wellcomecollection/wellcomecollection.org/pull/11652/commits/bf334465572b4f00dd8975ac512225706ac5b614)